c10/core/DeviceType.cpp

```diff
@@ -89,4 +89,18 @@ std::ostream& operator<<(std::ostream& stream, DeviceType type) {
   return stream;
 }
 
+DeviceType& _get_default_device_type() {
+  static DeviceType default_dev_type = DeviceType::CUDA;
```

Comment:
can we generate default device type by build script?

Reply:
The default device type can be changed thru the `set_default_device_type`.